posthog/urls.py

```diff
@@ -137,55 +142,76 @@ def signup_to_organization_view(request, invite_id):
+    user_name = details["fullname"]
+    strategy.session_set("user_name", user_name)
+    from_invite = False
     invite_id = strategy.session_get("invite_id")
-    if invite_id is None:
-        processed = render_to_string(
-            "auth_error.html",
-            {
-                "message": "There is no organization associated with this account! Please use an invite link from an organization to create an account!"
-            },
+    if not invite_id:
+        company_name = strategy.session_get("company_name", None)
+        email_opt_in = strategy.session_get("email_opt_in", None)
+        if not company_name or email_opt_in is None:
+            return redirect(finish_social_signup)
+        _, _, user = User.objects.bootstrap(
+            company_name=company_name, first_name=user_name, email=user_email, email_opt_in=email_opt_in, password=None
         )
-        return HttpResponse(processed, status=401)
-
-    fields = {name: kwargs.get(name, details.get(name)) for name in backend.setting("USER_FIELDS", ["email"])}
-
-    if not fields:
-        return
+    else:
+        from_invite = True
+        try:
+            invite: Union[OrganizationInvite, TeamInviteSurrogate] = OrganizationInvite.objects.select_related(
+                "organization"
+            ).get(id=invite_id)
+        except (OrganizationInvite.DoesNotExist, ValidationError):
+            try:
+                invite = TeamInviteSurrogate(invite_id)
+            except Team.DoesNotExist:
+                processed = render_to_string("auth_error.html", {"message": "Invalid invite link!"},)
+                return HttpResponse(processed, status=401)
 
-    try:
-        invite: Union[OrganizationInvite, TeamInviteSurrogate] = OrganizationInvite.objects.select_related(
-            "organization"
-        ).get(id=invite_id)
-    except (OrganizationInvite.DoesNotExist, ValidationError):
         try:
-            invite = TeamInviteSurrogate(invite_id)
-        except Team.DoesNotExist:
-            processed = render_to_string("auth_error.html", {"message": "Invalid invite link!"},)
+            invite.validate(user=None, email=user_email)
+        except ValueError as e:
+            processed = render_to_string("auth_error.html", {"message": str(e)},)
             return HttpResponse(processed, status=401)
 
-    try:
-        invite.validate(user=None, email=fields["email"])
-    except ValueError as e:
-        processed = render_to_string("auth_error.html", {"message": str(e)},)
-        return HttpResponse(processed, status=401)
+        try:
+            user = strategy.create_user(email=user_email, name=user_name)
+        except Exception as e:
+            processed = render_to_string(
+                "auth_error.html",
+                {
+                    "message": "Account unable to be created. This account may already exist. Please try again or use different credentials!"
```

Comment:
"Account unable to be created" sounds a bit weird

Reply:
That's some pretty old wording actually (and logic overall), I only put it in the `else` block. 🤔